vsifile/io/base.py

```diff
@@ -41,23 +50,43 @@ class BaseReader(metaclass=abc.ABCMeta):
     name: str = field()
```

Comment:
This reader class does overlap some with https://developmentseed.org/obstore/latest/api/file/#obstore.ReadableFile, which I'm trying to make as similar as possible to https://docs.python.org/3/library/io.html#io.BufferedReader. It doesn't look that bad to manage some of this file behavior in Python on this side, but some of it will get implemented in obstore natively.